### DISCUSSION TITLE
Is there a way to compute subsets of a full Jacobian?

### DISCUSSION THREAD

**Discussion by sea-bass** [stack-of-tasks/pinocchio#2287] [Q&A] [answered]
This is probably already possible, but I couldn't easily find this information in the docs.

Nominally, for an N degree-of-freedom robot model, your Jacobian would be 6-by-N.

But suppose we only want the Jacobian for a subset of those joints (or less importantly, even the degrees of freedom). Instead of computing the full Jacobian and then sub-slicing the resulting matrix, is there a way to more efficiently compute just the columns we care about and get a smaller matrix in the first place?

**Comment by jcarpent** [comment, marked as answer]
Yes, we can add this feature.
Currently, `computeJointJacobians` pre-compute all the quantities to extract the Jacobian for any joint or frame with `getJoinJacobian` or `getFrameJacobian`.
`computeJointJacobian(model,data,q,joint_id)` just returns the Jacobian for the given `joint_id`.

@sea-bass Have you encountered any computational burden with Jacobian computation?
I ask this question as normally, for robots such as humanoids, such operations are near the microseconds ;)

Would you like a function to compute the Jacobian of a given set of indexes, for instance?

**Reply by sea-bass** [reply]
Thanks! I have not encountered any computational burden; was more curious for the API convenience levels.

This is motivated by two use cases.

1. The need to have "joint groups", so for example we may want to get the Jacobian for joints 1-5 out of a 20 DOF system. (This is what you asked above, so yes)
2. Computing contact Jacobians using collision/distance results where we only ever care about the positional DOFs, so we would always be chopping off the last 3 rows.

**Reply by jcarpent** [reply]
We will add this feature then. The second point is already almost there ;)

**Reply by sea-bass** [reply]
Thank you! Happy to review/test as you see fit.
